rest/model/distro.go

```diff
@@ -18,6 +22,20 @@ func (apiDistro *APIDistro) BuildFromService(h interface{}) error {
 	case distro.Distro:
 		apiDistro.Name = ToAPIString(v.Id)
 		apiDistro.UserSpawnAllowed = v.SpawnAllowed
+		apiDistro.Provider = ToAPIString(v.Provider)
+
+		if v.ProviderSettings != nil {
+			ec2Settings := &cloud.EC2ProviderSettings{}
+			err := mapstructure.Decode(v.ProviderSettings, ec2Settings)
+			if err != nil || ec2Settings.AMI == "" {
```

Comment:
I would handle these 2 scenarios differently - if there is an error return it (it's not too important to manipulate data if something errors because returning an error to the caller means that other return values should not be trusted)

Reply:
Changed the parent if condition, now return the error if there is one